runner/s3_import_dynamodb.go

```diff
@@ -0,0 +1,233 @@
+
+package runner
+
+import (
+	"fmt"
+
+	"time"
+
+	"github.com/featureform/filestore"
+	"github.com/featureform/logging"
+	"github.com/featureform/metadata"
+	"github.com/featureform/provider"
+	pt "github.com/featureform/provider/provider_type"
+	"github.com/featureform/types"
+	"go.uber.org/zap"
+)
+
+type S3ImportMaterializationOption struct {
+	storeType  pt.Type
+	outputType filestore.FileType
+}
+
+func (o S3ImportMaterializationOption) Output() filestore.FileType {
+	return o.outputType
+}
+
+func (o S3ImportMaterializationOption) StoreType() pt.Type {
+	return o.storeType
+}
+
+func (o S3ImportMaterializationOption) ShouldIncludeHeaders() bool {
+	return false
+}
+
+type S3ImportDynamoDBRunner struct {
+	Online      provider.ImportableOnlineStore
+	Offline     provider.OfflineStore
+	OfflineType pt.Type
+	ID          provider.ResourceID
+	VType       provider.ValueType
+	IsUpdate    bool // Not currently useable
+	Logger      *zap.SugaredLogger
+}
+
+func (r S3ImportDynamoDBRunner) Resource() metadata.ResourceID {
+	return metadata.ResourceID{
+		Name:    r.ID.Name,
+		Variant: r.ID.Variant,
+		Type:    provider.ProviderToMetadataResourceType[r.ID.Type],
+	}
+}
+
+func (r S3ImportDynamoDBRunner) IsUpdateJob() bool {
+	return r.IsUpdate
+}
+
+func (r S3ImportDynamoDBRunner) Run() (types.CompletionWatcher, error) {
+	r.Logger.Infow("Staring S3 import to DynamoDB materialization runner", "name", r.ID.Name, "variant", r.ID.Variant)
+
+	if r.IsUpdate {
+		return nil, fmt.Errorf("materialization updates are not implemented for S3 import to DynamoDB")
+	}
+
+	option := S3ImportMaterializationOption{
+		storeType:  pt.SparkOffline,
+		outputType: filestore.CSV,
+	}
+
+	mat, err := r.Offline.CreateMaterialization(r.ID, option)
+	if err != nil {
+		r.Logger.Errorf("failed to create materialization: %v", err)
+		return nil, err
+	}
+
+	sparkOffline, ok := r.Offline.(*provider.SparkOfflineStore)
+	if !ok {
+		r.Logger.Errorf("offline store is not a SparkOfflineStore")
+		return nil, fmt.Errorf("offline store is not a SparkOfflineStore")
+	}
+
+	// **NOTE:** Unlike ResourceID, which has methods to convert the name, variant and type of resource to and from a path,
+	//  MaterializationID is a string that is already in the form of `/Materialization/<name>/<variant>`. We currently need
+	// to append `featureform/` to the materialization ID to get the source dir path, but this is not ideal. We should
+	// probably change the type of MaterializationID to be ResourceID.
+	sourceDirPath, err := sparkOffline.Store.CreateDirPath(fmt.Sprintf("featureform/%s", mat.ID()))
+	if err != nil {
+		r.Logger.Errorf("failed to create source dir path for resource %s: %v", r.ID.ToFilestorePath(), err)
+		return nil, err
+	}
+
+	files, err := sparkOffline.Store.List(sourceDirPath, filestore.CSV)
+	if err != nil {
+		r.Logger.Errorf("failed to list files in source dir path %s: %v", sourceDirPath, err)
+		return nil, err
+	}
```

Comment:
@sdreyer, I really wish I didn't have to break our abstractions to do this, but I need a file path from the newly created materialization. I _could_ change the `Materialization` interface to include a new method so that I don't have to rely on this type assertion. Thoughts?

Reply:
Discussed, not a great way to do this without a large refactor